core/src/main/kotlin/com/samples/verifier/internal/SamplesVerifierInstance.kt

```diff
@@ -115,6 +132,50 @@ internal class SamplesVerifierInstance(compilerUrl: String, kotlinEnv: KotlinEnv
     }
   }
 
+  private fun processDiffBranchesRepository(
+    baseUrl: String,
+    baseBranch: String,
+    headUrl: String,
+    headBranch: String,
+    type: FileType
+  ): RepoChanges {
+    val dir = File(baseUrl.substringAfterLast('/').substringBeforeLast('.'))
+    try {
+      logger.info("Cloning repository...")
+      cloneRepository(dir, baseUrl, baseBranch, true).use { git ->
+        logger.info("Getting diff between $baseUrl:$baseBranch and $headUrl:$headBranch")
+
+        val fr = fetch(git, headUrl, headBranch)
+        val newName = fr.getTrackingRefUpdates().first().localName
+
+        // git diff $(git-merge-base A B) B
+        // aka triple dots diff
+        val commonAncestor = mergeBase(git, baseBranch, newName)
+        val headCommit = getCommit(git.repository, newName)
+        val diff = diff(git, commonAncestor, headCommit)
+
+        val diffFilenames = getModifiedOrAddedFilenames(diff)
+        diffFilenames.forEach { logger.info("File $it is found in branches diff") }
+        val allFilenames = diffFilenames //+ filenames.orEmpty()
+        val diffInfo = DiffOfRepository(baseBranch, headBranch, getDeletedFilenames(diff))
+        return RepoChanges(diffInfo, processRepoFiles(git.repository, headCommit, allFilenames, type))
+      }
+    } catch (e: GitException) {
```

Comment:
I saw almost the same code in that file. Is it possible to avoid code duplicates?